lib/em-hiredis/pubsub_connection.rb

```diff
@@ -108,14 +111,23 @@ def handle_response(reply)
           @auth_df.succeed(reply)
         end
         @auth_df = nil
+      elsif @ping_df
+        if reply.kind_of?(Exception)
+          e = EM::Hiredis::RedisError.new(reply.message)
+          e.redis_error = reply
+          @ping_df.fail(e)
+        else
+          @ping_df.succeed(reply)
+        end
+        @ping_df = nil
```

Comment:
This concerns me because it relies on the user of this class to call "auth" and then "ping" in a specific order. If auth ping is called first, followed by auth, then it'll be the pong response which actually succeeds the `@auth_df`.

IMO this would be much easier to reason about if the same approach was used as the `RedisConnection` module, i.e. to have a `@response_queue` that are processed in the order the commands were sent.